src/ui/favoritespanewidget.cc

```diff
@@ -1040,12 +1033,11 @@ QModelIndex FavoritesModel::getModelIndexByFullPath( const QStringList & fullPat
       qsizetype rowIndex           = std::distance( childItems.begin(), folder_found );
       QModelIndex found_modelIndex = createIndex( rowIndex, 0, *folder_found );
 
-      if ( pathPart == fullPath.end() - 1 ) {
+      if ( pathPart == fullPath.back() ) {
```

Comment:
This method's change should be similar to getItemByFullPath 
instead of just change iterator to range for @xiaoyifang .

back() is the last data.

The end()-1 is the last index.

For example, when the target path is "a/c/b"

The folder  "a/b" will be return wrongly.

The iterator and range for are not exactly the same.

Reply:
you are right.

Reply:
#2207